rs/execution_environment/src/scheduler.rs

```diff
@@ -700,19 +700,32 @@ impl SchedulerImpl {
 
             // Update subnet available memory before taking out the canisters.
             round_limits.subnet_available_memory = self.exec_env.subnet_available_memory(&state);
-            let canisters = state.take_canister_states();
-            // Obtain the active canisters and update the collection of heap delta rate-limited canisters.
-            let (active_round_schedule, rate_limited_canister_ids) = round_schedule
-                .filter_canisters(
-                    &canisters,
-                    self.config.heap_delta_rate_limit,
-                    self.rate_limiting_of_heap_delta,
+            let mut canisters = state.take_canister_states();
+
+            // Scheduling.
```

Comment:
High-level comment: Does this change help move us closer to the quick fix? (I.e. does it make it easier to charge all canisters that got a chance at a full round or not?)

Because OTOH, we have both subnets that spend [15-20 ms scheduling](https://grafana.mainnet.dfinity.network/explore?orgId=1&left=%7B%22datasource%22:%22P119963096DF10E67%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22P119963096DF10E67%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22label_replace%28%5Cn%20%20quantile%20by%20%28ic,%20ic_subnet,%20phase%29%20%28%5Cn%20%20%20%200.5,%5Cn%20%20%20%20%20%20rate%28execution_round_scheduling_duration_seconds_sum%7Bic%3D%5C%22mercury%5C%22%7D%5B1m%5D%29%5Cn%20%20%20%20%2F%5Cn%20%20%20%20%20%20rate%28mr_process_batch_duration_seconds_count%7Bic%3D%5C%22mercury%5C%22%7D%5B1m%5D%29%5Cn%20%20%29,%5Cn%20%20%5C%22ic_subnet%5C%22,%20%5C%22$1%5C%22,%20%5C%22ic_subnet%5C%22,%20%5C%22%28%5Ba-z0-9%5D%2B%29-.%2A%5C%22%5Cn%29%22,%22legendFormat%22:%22__auto%22,%22range%22:true,%22instant%22:true%7D%5D,%22range%22:%7B%22from%22:%221727850252996%22,%22to%22:%221727853852019%22%7D%7D); and subnets that do [12 inner loop iterations](https://grafana.mainnet.dfinity.network/d/execution-metrics/execution-metrics?from=1727850037744&to=1727853637744&var-datasource=IC%20Metrics%20%28cluster%20local%29&var-ic=mercury&var-ic_subnet=All&var-dc=All&var-instance=All&var-instance_host=All&var-period=$__auto_interval_period&orgId=1&viewPanel=136&var-node_instance=All&var-heatmap_period=$__auto_interval_heatmap_period&editPanel=136) per round. Luckily no subnet happens to be in both those groups, but even assuming no increase in these numbers, there's nothing stopping a subnet from doing 250 ms worth of scheduling out of a 400 ms round.

Reply:
This change shifts the schedule by four canisters per inner round, i.e. potentially 12 times faster. I'm aware of the potential performance impact but have prioritized other optimizations for the upcoming release...